quarkchain/cluster/subscription.py

```diff
@@ -57,14 +57,18 @@ def remove_subscriber(self, sub_id):
     async def notify_sync(
         self, running: bool, tip: MinorBlockHeader, queue: List[MinorBlockHeader]
     ):
-        # TODO
-        data = None
+        data = [running, tip, queue]
         await self.__notify(SUB_SYNC, data)
 
     async def __notify(self, sub_type, data):
         assert sub_type in self.subscribers
         for sub_id, websocket in self.subscribers[sub_type].items():
-            response = self.response_encoder(sub_id, data)
+            if sub_type == SUB_SYNC:
+                response == self.syncing_response_encoder(
```

Comment:
Maybe the data encoding process can be put in notify_sync? I guess this can make __notify() look cleaner.

Reply:
We need to call `syncing_response_encoder ` specifically for syncing because of its different response format from other events.